images/router/haproxy/conf/haproxy-config.template

```diff
@@ -226,6 +226,11 @@ backend be_edge_http_{{$cfgIdx}}
     {{ else }}
   balance leastconn
     {{ end }}
+    {{ with $value := index $cfg.Annotations "router.openshift.io/haproxy.timeout"}}
+      {{if (matchString "[0-9]*(us|ms|s|m|h|d)?" $value) }}
```

Comment:
I assume matchString has an implicit anchor to the start and end of the string?  But can we check that "test 9ms test" doesn't match.

Also * matches 0+ characters... I assume you want [0-9]+ to make it match 1+.  Also \d is usually a synonym for [0-9], so let's see if "\d+(us|ms|s|m|h|d)?" works.